### ISSUE TITLE
aws-elasticache: extraneous key [Tags] is not permitted on CfnUserGroup

### ISSUE THREAD

**Issue by ThomasSteinbach** [aws/aws-cdk#25409]
### Describe the bug

Between aws-cdk-lib in version 2.54.0 and 2.77.0 the tagging of the Resource AWS::ElastiCache::UserGroup was introduced. However since the update of the aws-cdk-lib our deployments (of the Elasticache UserGroups) fail with:

```
Properties validation failed for resource abcdgroup2CA6EC0B with message: #: extraneous key [Tags] is not permitted
```

This is what the CFN changes after the update of the aws-cdk-lib look like:

```
  testcacheapplicationgroup822A7DA9:
    Type: AWS::ElastiCache::UserGroup
    Properties:
      Engine: redis
      UserGroupId: testapp-testenv-testsystem-cdk-11728616
      UserIds:
      - default28cd8437
      - testuser11b4cdcd
+     Tags:
+     - Key: TagOne
+       Value: valueone
+     - Key: TagTwo
+       Value: valuetwo
```

### Expected Behavior

The CDK deployment would also work after the update of the aws-cdk-library from 2.54 to 2.77

### Current Behavior

The CDK/CFN-Deployment fails with the error shown in the description.

### Reproduction Steps

* create a `aws_cdk.aws_elasticache.CfnUserGroup`
* tag this construct or the whole stack with `aws_cdk.Tags.of(sope).add(...)`

### Possible Solution

Our current workaround is to exclude the resource from EVERY tag aspect in the whole CDK code:

```python
aws_cdk.Tags.of(self).add("Environment", "dev", exclude_resource_types=["AWS::ElastiCache::UserGroup"])
```

### CDK CLI Version

2.77.0

### Node.js Version

v18.15.0

### OS

LInux / Mac

### Language

Python

**Comment by peterwoodworth** [comment]
I'm not able to reproduce this, my tags get set on the usergroup on deploy. Here's a simple stack that deploys on latest version:

```ts
export class TestingStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    new elasticache.CfnUserGroup(this, 'usergroup', {
      engine: 'redis',
      userGroupId: 'usergroup',
      userIds: ['default'],
    });
  }
}

const stack = new TestingStack(app, 'ElasticacheStack', {
  env: { account: process.env.CDK_DEFAULT_ACCOUNT, region: 'us-east-1' },
});

cdk.Tags.of(stack).add('MyKey', 'MyValue');
```

If my understanding is correct, you added this tag to your stack/construct that creates this resource before usergroup was considered taggable, then when you updated to `2.77.0` from `2.54.0` you ran into this error because it started trying to add the tags to the usergroup?

**Comment by peterwoodworth** [comment]
I'm not able to reproduce with the code you've provided still. If you're able to find a way to reproduce this consistently, and share it in a GitHub repository that I can easily clone and run, then please do that and I might be able to provide additional help. Else, I recommend going through premium support if you have it. This _can_ be tagged, so there's nothing we would fix here

**Comment by peterwoodworth** [comment]
Actually, I didn't try deploying to the same region. Let me check if this is available in `eu-central-1`

**Comment by peterwoodworth** [comment]
Yeah this isn't available in `eu-central-1`. Maybe this is something we can add to `RegionInfo`? I'm not sure though. For now, I recommend [excluding the CfnUserGroup resource type](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_core.TagProps.html#excluderesourcetypes)

**Comment by ThomasSteinbach** [comment]
# Workaround

Since it could take years a solution is in sight, the [best temporary solution is not to tag CfnUserGroups](https://github.com/aws/aws-cdk/issues/25409#issuecomment-1555156342).
Add this aspect to your stack:

```python
@jsii.implements(aws_cdk.IAspect)
class TagRemoverAspect:
    def visit(self, node):
        # Per default we exclude following resource types from tagging:
        # * Elasticache::UserGroup - has the error: "extraneous key [tags] is not
        if isinstance(node, aws_cdk.aws_elasticache.CfnUserGroup):
            node.add_deletion_override("Properties.Tags")

aws_cdk.Aspects.of(self).add(TagRemoverAspect())
```

This solution is a more universal approach than [excuding tags via the Tags Manager](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.TagProps.html#excluderesourcetypes), because you maybe don't know where in your constructs hierarchy you have applied tags.

The aspect will take care, that the CfnUserGroup didn't receive any tags from your code.
No tags, no CFN problems.
